pkg/build/controller/strategy/util.go

```diff
@@ -649,3 +653,105 @@ func setupBlobCache(pod *corev1.Pod) {
 		pod.Spec.InitContainers = initContainers
 	}
 }
+
+// injectBuildVolumesIntoPod injects the provided BuildVolumes into given Pod
+func addBuildVolumesToPod(pod *corev1.Pod, buildVolumes []buildv1.BuildVolume) error {
+	if len(buildVolumes) == 0 {
+		return nil
+	}
+
+	for _, bv := range buildVolumes {
+		v := corev1.Volume{
+			Name: bv.Name,
+		}
+
+		switch {
+		case bv.Type == buildv1.BuildVolumeSourceTypeSecret:
+			v.Secret = bv.Secret
+		case bv.Type == buildv1.BuildVolumeSourceTypeConfigMap:
+			v.ConfigMap = bv.ConfigMap
+		default:
+			return fmt.Errorf("encountered unsupported build volume source type %q", bv.Type)
+		}
+
+		pod.Spec.Volumes = append(pod.Spec.Volumes, v)
+	}
+
+	return nil
+}
+
+// injectBuildVolumeMountsIntoContainer injects the provided BuildVolumeMounts into the given Container
+func addBuildVolumeMountsToContainer(container *corev1.Container, buildVolumeMounts []buildv1.BuildVolumeMount) error {
+	if len(buildVolumeMounts) == 0 {
+		return nil
+	}
+
+	for _, bvm := range buildVolumeMounts {
+		vm := corev1.VolumeMount{
+			Name:      bvm.Name,
+			ReadOnly:  bvm.ReadOnly,
+			MountPath: path.Join(BuildVolumeMountPath, bvm.MountPath),
+		}
+
+		container.VolumeMounts = append(container.VolumeMounts, vm)
+	}
+
+	return nil
+}
+
+// setupBuildVolumes sets up user defined BuildVolumes and BuildVolumeMounts
+func setupBuildVolumes(pod *corev1.Pod, buildVolumes []buildv1.BuildVolume, buildVolumeMounts []buildv1.BuildVolumeMount) error {
+	// if there are no BuildVolumes or BuildVolumeMounts, or the pod is nil,
+	// there is no processing needed, so just return quickly
+	if (len(buildVolumes) == 0 && len(buildVolumeMounts) == 0) || pod == nil {
+		return nil
+	}
+
+	// generate a map of already used volume names and mounts in the container
+	used := make(map[string]struct{})
+
+	// iterate over existing Volumes and add to the map
+	for _, v := range pod.Spec.Volumes {
+		used[v.Name] = struct{}{}
+	}
+
+	// check the buildVolumes for collisions
```

Comment:
Add a TODO here - we can probably do better.

```suggestion
	// check the buildVolumes for collisions
	// TODO: Generate unique names or a numbering scheme to avoid volume name collisions
```

Reply:
updated in next push